packages/joint-core/src/dia/groups/CellGroup.mjs

```diff
@@ -0,0 +1,67 @@
+import { Model, Collection } from '../../mvc/index.mjs';
+
+export class CellGroupCollection extends Collection {
+
+    _prepareModel(attrs, _options) {
+        if (this._isModel(attrs)) {
+            // do not change collection attribute of a cell model
+            return attrs;
+        } else {
+            throw new Error('CellGroupCollection only accepts Cell instances.');
+        }
+    }
+
+    _onModelEvent(event, model, collection, options) {
+        if (model) {
+            if (event === 'changeId') {
```

Comment:
You can refactor `Collection.mjs` if you need to partially  preserved its behavious.

Why do we omit this line?
```ts
if ((event === 'add' || event === 'remove') && collection !== this) return;
```

Reply:
Added a comment. It is done, so the CellGroup can emit `add` and `remove` events, because the `collection` is normally `graph.cellCollection` origin collection